src/apps/explorer/pages/AppData/FormPage.tsx

```diff
@@ -0,0 +1,267 @@
+import { Notification } from 'components/Notification'
+import {
+  INITIAL_FORM_VALUES,
+  INVALID_IPFS_CREDENTIALS,
+  getSchema,
+  transformErrors,
+  handleErrors,
+  deletePropertyPath,
+  ipfsSchema,
+  uiSchema,
+  ipfsUiSchema,
+  CustomField,
+  FormProps,
+} from './config'
+import { IpfsWrapper } from './styled'
+import { ExternalLink } from 'components/analytics/ExternalLink'
+
+const FormPage: React.FC = () => {
+  const [schema, setSchema] = useState<JSONSchema7>({})
+  const [appDataForm, setAppDataForm] = useState({})
+  const [disabledAppData, setDisabledAppData] = useState<boolean>(true)
+  const [disabledIPFS, setDisabledIPFS] = useState<boolean>(true)
+  const [invalidFormDataAttempted, setInvalidFormDataAttempted] = useState<{ appData: boolean; ipfs: boolean }>({
+    appData: false,
+    ipfs: false,
+  })
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [ipfsHashInfo, setIpfsHashInfo] = useState<IpfsHashInfo | void | undefined>()
+  const [ipfsCredentials, setIpfsCredentials] = useState<{ pinataApiKey?: string; pinataApiSecret?: string }>({})
+  const [isDocUploaded, setIsDocUploaded] = useState<boolean>(false)
+  const [error, setError] = useState<string>()
+
+  const network = useNetworkId()
+  const formRef = React.useRef<Form<FormProps>>(null)
+  const ipfsFormRef = React.useRef<Form<FormProps>>(null)
+
+  useEffect(() => {
+    const fetchSchema = async (): Promise<void> => {
+      const latestSchema = await getSchema()
+      setSchema(latestSchema)
+      setAppDataForm(INITIAL_FORM_VALUES)
+    }
+
+    fetchSchema()
+  }, [])
+
+  const toggleInvalid = (data: { [key: string]: boolean }): void => {
+    setInvalidFormDataAttempted((prevState) => ({ ...prevState, ...data }))
+  }
+
+  const resetFormFields = (form: string): void => {
+    setDisabledIPFS(true)
+    setIpfsCredentials({})
+    toggleInvalid({ ipfs: false })
+    if (form === 'appData') {
+      setDisabledAppData(true)
+    }
+  }
+
+  const handleFormatData = (formData: any): any => {
+    if (!formData.metadata || !Object.keys(formData.metadata).length) return formData
+    const formattedData = structuredClone(formData)
+    const isReferrerEnabled = formattedData.metadata.referrer.enableReferrer
+    const isQuoteEnabled = formattedData.metadata.quote.enableQuote
+
+    deletePropertyPath(formattedData, 'metadata.referrer.enableReferrer')
+    deletePropertyPath(formattedData, 'metadata.quote.enableQuote')
+
+    if (!isReferrerEnabled) {
+      deletePropertyPath(formattedData, 'metadata.referrer')
+    }
+    if (!isQuoteEnabled) {
+      deletePropertyPath(formattedData, 'metadata.quote')
+    }
+
+    return formattedData
+  }
+
+  const handleMetadataErrors = (_: FormProps, errors: FormValidation): FormValidation =>
+    handleErrors(formRef, errors, setDisabledAppData)
+
+  const handleIPFSErrors = (_: FormProps, errors: FormValidation): FormValidation =>
+    handleErrors(ipfsFormRef, errors, setDisabledIPFS)
```

Comment:
`useCallback` on the handlers? Most of the inner functions in this component actually